src/NHibernate.Test/Async/Linq/NullComparisonTests.cs

```diff
@@ -560,6 +577,15 @@ public async Task NullInequalityAsync()
 
 			await (ExpectAsync(session.Query<User>().Where(o => o.CreatedBy.ModifiedBy.Id != 5), Does.Contain("is null").IgnoreCase));
 			await (ExpectAsync(session.Query<User>().Where(o => 5 != o.CreatedBy.ModifiedBy.Id), Does.Contain("is null").IgnoreCase));
+
+			short value = 3;
```

Comment:
unnecessary cast is exists short/long with == / <= / >= operators could you add additional test cases.

Reply:
When comparing two different types `short == long`, a cast will always be added when the mapped sql type of `short` and `long` are different. I am thinking of restoring the previous behavior of `Equals` in case a user wants to compare two different types without an explicit cast as this is not possible anymore in `5.3`.

Reply:
I think the operators and methods should work exactly the same, otherwise the software developer might make an error about which one to use and under what condition. If not careful, sql clauses are created which can cause performance problems.

The strange thing is that it casts unnecessarily in short to short or long to long comparisons.

Equals method with == operator 
CompareTo method result == 0 with <=  operator
CompareTo method result < 0 with <= operator
CompareTo method result > 0 with >= operator